cmd/cosign/cli/fulcio/fulcio.go

```diff
@@ -69,66 +55,37 @@ func (rf *realConnector) OIDConnect(url, clientID, secret string) (*oauthflow.OI
 	return oauthflow.OIDConnect(url, clientID, secret, rf.flow)
 }
 
-type signingCertProvider interface {
-	SigningCert(params *operations.SigningCertParams, authInfo runtime.ClientAuthInfoWriter, opts ...operations.ClientOption) (*operations.SigningCertCreated, error)
-}
-
-func getCertForOauthID(priv *ecdsa.PrivateKey, scp signingCertProvider, connector oidcConnector, oidcIssuer string, oidcClientID string) (Resp, error) {
+func getCertForOauthID(priv *ecdsa.PrivateKey, fc api.Client, connector oidcConnector, oidcIssuer string, oidcClientID string) (*api.CertificateResponse, error) {
 	pubBytes, err := x509.MarshalPKIXPublicKey(&priv.PublicKey)
 	if err != nil {
-		return Resp{}, err
+		return nil, err
 	}
 
 	tok, err := connector.OIDConnect(oidcIssuer, oidcClientID, "")
 	if err != nil {
-		return Resp{}, err
+		return nil, err
 	}
 
 	// Sign the email address as part of the request
 	h := sha256.Sum256([]byte(tok.Subject))
 	proof, err := ecdsa.SignASN1(rand.Reader, priv, h[:])
 	if err != nil {
-		return Resp{}, err
+		return nil, err
 	}
 
-	bearerAuth := httptransport.BearerToken(tok.RawString)
-
-	content := strfmt.Base64(pubBytes)
-	signedChallenge := strfmt.Base64(proof)
-	params := operations.NewSigningCertParams()
-	params.SetCertificateRequest(
-		&models.CertificateRequest{
-			PublicKey: &models.CertificateRequestPublicKey{
-				Algorithm: models.CertificateRequestPublicKeyAlgorithmEcdsa,
-				Content:   &content,
-			},
-			SignedEmailAddress: &signedChallenge,
+	cr := api.CertificateRequest{
+		PublicKey: api.Key{
+			Algorithm: "ecdsa",
```

Comment:
should make this a custom type and/or enum imho

Reply:
Want to send a PR to Fulcio?

Reply:
will do